File: internal/util/ExecTerminalScript.go
package util

import (
	"os"
	"os/exec"
)

func ExecTerminalScript(script string) error {
	scriptFile, err := os.CreateTemp("", "kegger")
	if err != nil {
		return err
	}

	err = scriptFile.Chmod(0775)
	if err != nil {
		return err
	}

	_, err = scriptFile.WriteString(script)
	if err != nil {
		return err
	}
	scriptFile.Sync()

	cmd := exec.Command("open", "-b", "com.apple.terminal", scriptFile.Name())
	err = cmd.Run()

	return err
}

File: internal/util/execTerminalScript.js
const updateScript = "./bin/update.sh";

export default async function execTerminalScript(script) {
  const tempScript = await Deno.makeTempFile({
    prefix: "kegger",
    suffix: ".sh",
  });
  await Deno.copyFile(updateScript, tempScript);
  await Deno.writeTextFile(tempScript, script);

  const terminalProcess = Deno.run({
    cmd: ["open", "-b", "com.apple.terminal", tempScript],
    env: {},
  });

  const status = await terminalProcess.status();
  // Deno.remove(tempScript);

  return status;
}
